automation/android/espresso/build.gradle

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2017 - present Instructure, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+buildscript {
+    // Define this once here because it will need to be reused later
+    ext.kotlinVersion = "1.2.70"
+
+    repositories {
+        jcenter()
+        google()
+    }
+
+    dependencies {
+        classpath "com.android.tools.build:gradle:3.2.1"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
+    }
+}
+
+allprojects {
+    repositories {
+        jcenter()
+        google()
+    }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+// Borrowed from http://stackoverflow.com/questions/17097263/automatically-versioning-android-project-from-git-describe-with-android-studio-g
+def getVersionCode = {
```

Comment:
My attempt at automatically producing a version number.

Reply:
we may want to consistently use the commit hash, I'm not sure there's an easy way to resolve build numbers to git commits.

Reply:
This has worked for me in the past.  And commit hashes wouldn't be ordered, right?

Reply:
I guess that my thinking is that we'll tag each bitrise build with the bitrise build number, and that will be used to identify whatever we push to jitpack.  I suppose that we could manually maintain the build number, like we do with our canvas apps, but I'd like to see if this works.

Reply:
The hash becomes the build number. Checkout how Java projects publish snapshots on jitpack. It's usually something like Flank-master-f40414091f-1.jar

Reply:
> And commit hashes wouldn't be ordered, right?

they're ordered in the git history, that's about it. GitHub will show you how many commits behind a commit is.

Reply:
Apologies for my slowness, but how is that better than Flank-master-1, Flank-master-2, Flank-master-3, ...?

Reply:
let's chat more on Slack, I think there are a few reasons why that's not common practice on jitpack.